biz.aQute.bndlib/src/aQute/bnd/osgi/Macro.java

```diff
@@ -356,6 +354,24 @@ private String doCommand(Object target, String method, String[] args) {
 		return null;
 	}
 
+	private Method lookupMacroMethod(String method, Object target) {
+		// Assume macro names do not start with '-' and does not contain '.'
+		if (method.startsWith("-") || method.indexOf('.') > -1) {
+			return null;
+		}
+		String cname = '_' + method.replace('-', '_');
+		for (Method m : target.getClass().getMethods()) {
+			if (m.getParameterTypes().length != 1) {
+				continue;
+			}
+			if (m.getName().equals(cname)) {
+				if (m.getParameterTypes()[0].getName().equals(String[].class.getName()))
```

Comment:
Why not `if (m.getParameterTypes()[0] == String[].class)`. Comparing class names seems wrong.